src/components/ListOfCards.js

```diff
@@ -2,24 +2,29 @@ import React, { Component } from 'react';
 import styled from 'styled-components'
 import Card from "./Card";
 import { getKudoses } from '../api/KudosApi'
+import Spinner from "./Spinner";
 
 class ListOfCards extends Component {
     state = {
-        kudoses: []
+        kudoses: [],
+        loading: true
     }
     componentWillMount () {
         getKudoses()
             .then(kudoses => {
                 this.setState({
-                    kudoses
+                    kudoses,
+                    loading: false
                 })
             })
     }
     render() {
+        const { loading, kudoses } = this.state
         return (
             <StyledList>
                 {
-                    this.state.kudoses.map((kudos, index) => <Card kudos={kudos} key={index}/>)
+                    loading ? <Spinner/> :
+                    kudoses.map((kudos, index) => <Card kudos={kudos} key={index}/>)
```

Comment:
it is a good practice to use real ID instead of index. Thanks to that, React will know if the given element is the same element or a new one.
```
// array like
<Card key={0}/>
<Card key={1}/>

// ID like
<Card key={'34eqr4tzcv-random-id-which-may-be-a-string-or-number'}/>
```

https://reactjs.org/docs/lists-and-keys.html#keys

```We don’t recommend using indexes for keys if the order of items may change. This can negatively impact performance and may cause issues with component state.```

Reply:
What if I make key like a combination of kudos 'name', 'description' and 'from' property (it's all data which I have from backend), and someone add one kudos more than once? Keys will be duplicated. Maybe we could add an ID property to object to make keys combination unique?

Reply:
@ulaulala backend shall return you the ID; as you noticed making self-constructed ID will not grant the uniquity

Reply:
Ok, I will make an issue for that